sktime/base/_base.py

```diff
@@ -12,17 +12,48 @@
 from sktime.exceptions import NotFittedError
 
 
-class BaseEstimator(_BaseEstimator):
-    """Base class for defining estimators in sktime. Extends scikit-learn's
-    BaseEstimator.
+class BaseObject(_BaseEstimator):
+    """Base class for defining other classes in sktime.
+
+    Extends scikit-learn's BaseEstimator to include sktime interface for tags.
+    """
+
+    @classmethod
+    def _all_tags(cls):
+        """Get tags from estimator class and all its parent classes.
+
+        Creates a separate sktime tag interface in addition to the one in
+        scikit-learn to make sure it does not interfere with scikit-learn's tag
+        interface when inheriting from scikit-learn classes. Sktime's
+        estimator tags are class rather than object attribute as in scikit-learn.
+        """
```

Comment:
this should properly describe that this implements a kind of inheritance, no?
As far as I understand, if multiple estimators in the inheritance tree have the same tag, the descendant tag overrides the parent tag.
This is important to document, perhaps with an example of what tags are returned.

Reply:
I'd also like to rename the `_all_tags` method to `get_tags` and make it public if that doesn't interfere with scikit-learn's tags. But perhaps changes to the tag system are best addressed in a separate PR.

Reply:
minor comment: I was wondering, upon a vague feeling, whether this could be just solved by using inheritance smartly? Namely, if we were just looking at the dictionary, and its keys as attribute names, this is precisely inheritance of attributes.

Inside the list, the name for this pattern is inheritance of nested attributes - it doesn't seem python supports this natively. A quick google search indicates it's an occasional topic in various other programming languages as well (mostly ending in the conclusion that the language doesn't support it).